File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/mission/model/MissionId.java
package com.study.webflux.rag.domain.mission.model;

import java.util.UUID;

public record MissionId(
	String value
) {
	public MissionId {
		if (value == null || value.isBlank()) {
			throw new IllegalArgumentException("missionId cannot be null or blank");
		}
	}

	public static MissionId of(String value) {
		return new MissionId(value);
	}

	public static MissionId generate() {
		return new MissionId(UUID.randomUUID().toString());
	}
}

File: domain/src/main/kotlin/com/miyou/app/domain/mission/model/MissionId.kt
package com.miyou.app.domain.mission.model

import java.util.UUID

data class MissionId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "missionId cannot be null or blank" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): MissionId = MissionId(value)

        @JvmStatic
        fun generate(): MissionId = MissionId(UUID.randomUUID().toString())
    }

    fun value(): String = value
}
